### PR TITLE
Update email-anonymizer.py & readme

### PR DESCRIPTION
**Pull request by titouan13**
Hi - I fixed some typos, added some doc around HMAC algorithm, fixed the regex & hmac for email-anonymizer.py

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -416,8 +416,8 @@ Configuration parameters:
 | Property         | Default value | Required | Description |
 |------------------|---------------|----------|-------------|
 | `salt`           | none          | **yes**  | HMAC key (salt) used for calculating the checksum |
-| `prefix`         | `<`           | no       | Prefix of emails in the log (for easier grepping) |
-| `suffix`         | `>`           | no       | Suffix of emails in the log (for easier grepping) |
```
Comment (bokysan) [inline README.md:420]: Why was this removed?

File: scripts/email-anonymizer.py [hunk 2]
```diff
@@ -33,7 +33,7 @@
 # This pattern below, should, however match anything that remotely looks like an email.
 # It is too broad, though, as it will match things which are not considered valid email
 # addresses as well. But for our use case, that's OK and more than sufficient.
-EMAIL_CATCH_ALL_PATTERN = '([^ "\\[\\]<>]+|".+")@(\[([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+|[A-Za-z0-9]+:.+)\]|([^ \\{}():;]+(\.[^ \\{}():;]+)*))'
+EMAIL_CATCH_ALL_PATTERN = '([^ "\\[\\]<]+|".+")@(\[([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+|[A-Za-z0-9]+:.+)\]|([^ \\{}():;>]+(\.[^ \\{}():;]+)*))'
```
Comment (bokysan) [inline scripts/email-anonymizer.py:36]: Why was this regex changed?

**Review by bokysan — COMMENTED** [review]
Hello,

thank you for this lovely commit! Changes like these are always appreciated!

Could you be so kind, though, to provide answers to the questions above before I finally merge this?

File: scripts/email-anonymizer.py (hunk 2 above)
Comment (titouan13) [reply scripts/email-anonymizer.py:36]: Hi - this regex was updated because it was catching the email address with a trailing '>'
e.g. "to=<test@yahoo.com>," was parsed to "test@yahoo.com>"

**Comment by bokysan** [comment]
Works, thank you for the commit!
